handler.go

```diff
@@ -129,6 +129,10 @@ func (h *baseHandler) broadcastName(room string) string {
 }
 
 func (h *socketHandler) onPacket(decoder *decoder, packet *packet) ([]interface{}, error) {
+	defer func() {
```

Comment:
Why do you use an unnecessary anonymous function instead of defer decoder.Close() ? And check for nil needed, because pointer used.